examples/src/main/java/ConvenienceExamples.java

```diff
@@ -0,0 +1,190 @@
+
+                // check remediation options to continue the flow
+                remediationOptions = idxResponse.remediation().remediationOptions();
+                remediationOptionsOptional = Arrays.stream(remediationOptions)
+                        .filter(x -> "challenge-authenticator".equals(x.getName()))
+                        .findFirst();
+                remediationOption = remediationOptionsOptional.get();
+
+                // answer password authenticator challenge
+                Credentials credentials = new Credentials();
+                credentials.setPasscode(password.toCharArray());
+
+                // build answer password authenticator challenge request
+                AnswerChallengeRequest passwordAuthenticatorAnswerChallengeRequest = AnswerChallengeRequestBuilder.builder()
+                        .withStateHandle(stateHandle)
+                        .withCredentials(credentials)
+                        .build();
+                idxResponse = remediationOption.proceed(client, passwordAuthenticatorAnswerChallengeRequest);
+
+                if (idxResponse.isLoginSuccessful()) {
+                    log.info("Login Successful!");
+                    tokenResponse = idxResponse.getSuccessWithInteractionCode().exchangeCode(client, idxClientContext);
+                    log.info("Exchanged interaction code for token: \naccessToken: {}, \nidToken: {}, \nrefreshToken: {}, \ntokenType: {}, \nscope: {}, \nexpiresIn:{}",
+                            tokenResponse.getAccessToken(),
+                            tokenResponse.getIdToken(),
+                            tokenResponse.getRefreshToken(),
+                            tokenResponse.getTokenType(),
+                            tokenResponse.getScope(),
+                            tokenResponse.getExpiresIn());
+                } else if (idxResponse.getMessages() != null) {
+                    Arrays.stream(idxResponse.getMessages().getValue()).forEach(msg -> log.error("Error: {}", msg.getMessage()));
+                } else {
+                    // authentication not successful yet; we need to follow more remediation steps
+                    log.info("Could not authenticate with just the provided username and password (password authenticator). More remediation steps needed...");
+                }
+            }
+        } catch (ProcessingException e) {
+            log.error("Something went wrong!", e);
+        } catch (IllegalArgumentException e) {
+            log.error("Exception occurred" ,e);
+        }
+
+        return tokenResponse;
+    }
+
+    public static void main(String... args) {
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("Enter Username: ");
+        String username = scanner.nextLine();
+        System.out.print("Enter Password: ");
+        String password = scanner.nextLine();
+
+        authenticate(username, password);
```

Comment:
Nit: We can add logging here instead of the wrapper method 
```
TokenResponse tokenResponse = authenticate(username, password);
log.info("Login Successful!");
log.info("Token: {}", tokenResponse);
```

Reply:
done